pkg/api/http_server.go

```diff
@@ -563,6 +568,7 @@ func (hs *HTTPServer) mapStatic(m *web.Mux, rootDir string, dir string, prefix s
 			SkipLogging: true,
 			Prefix:      prefix,
 			AddHeaders:  headers,
+			IndexFile:   "index.html",
```

Comment:
Does this means all non-existing directories will try to load index.html rather than returning a 404? Maybe make this configurable?

Reply:
checking closer this is redundant because we always set it if it's missing:https://github.com/grafana/grafana/blob/fd14112dbd93c97509f73e623b1a8234f012cb1d/pkg/api/static/static.go#L92-L94
but it didn't used to work until I made [this fix](https://github.com/grafana/grafana/pull/44541/files#diff-e13dbb34de534ed593d3bfb480d30b2640e0f4a730495d54575c52805b26083e)
so I've just removed it.

TBH it's hard to me to follow how `staticHandler()` is supposed to work in case of a directory; that's why I wanted some extra 👀  on it.
However, I tried to access an **existing** directory (public/dashboards) and I get 404 both in main and in this branch.

Reply:
Seems to work as expected